### PR TITLE
Add "no-curly-quotes" rule to the codebase. 

### PR DESCRIPTION
**Pull request by EliSchleifer**
Adds support for nocurlyquotes that will find instances of curlyquotes in a file and replace with correct non-curly form. We had instances of this in our code base and it caused some issues with marketing site. 

Fun little rule. Interestingly - this rule exposes a bug in how we apply on `trunk check` sarif fixes for utf-8 non-ascii characters. For now this rule will stay disabled until we have a fix.

Fix bug where multiple fixes in a single issue were being turned into multiple issues (which is not what the spec or code was intending to do).

### CODE REVIEW

File: src/rules/no_curly_quotes.rs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    let config = &run.config.nocurlyquotes;
+
+    if !config.enabled {
+        trace!("'nocurlyquotes' is disabled");
+        return Ok(vec![]);
+    }
+
+    if run.is_upstream() {
+        return Ok(vec![]);
+    }
```
Comment (TylerJang27) [inline src/rules/no_curly_quotes.rs:25]: What's wrong with running this on the upstream and having it behave like a normal linter? This shouldn't need upstream awareness like ICTC does...

File: src/rules/no_curly_quotes.rs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                },
+                inserted_content: rchar.to_string(),
+            })
+            .collect();
+
+        ret.push(Diagnostic {
+            path: path.to_str().unwrap().to_string(),
+            range: Some(Range {
+                start: Position {
+                    line: i as u64,
```
Comment (TylerJang27) [inline src/rules/no_curly_quotes.rs:94]: Are the replacements 1-indexed and the diagnostics are 0-indexed?

Comment (TylerJang27) [reply src/rules/no_curly_quotes.rs:94]: If not, the test you added asserts the message, not the diagnostic location

**Review by TylerJang27 — COMMENTED** [review]
Couple notes. Sanity checking as well that this is intended to be enabled for all files and then users can ignore ranges as desired?

File: src/rules/no_curly_quotes.rs (hunk 2 above)
Comment (EliSchleifer) [reply src/rules/no_curly_quotes.rs:94]: good catch - I shoud standardize this in the diagnostics.rs file where we make every 1 index assuming input is 0 except for startLine and endLine...whoops.

File: src/rules/no_curly_quotes.rs (hunk 1 above)
Comment (EliSchleifer) [reply src/rules/no_curly_quotes.rs:25]: I had disabled that for testing - good catch
shoudve put in a donotand

**Review by TylerJang27 — APPROVED** [review]

**Pull request merged**